src/index.ts

```diff
@@ -1,93 +1,167 @@
+  contacts: string;
+};
+
 class School {
   // implement 'add area', 'remove area', 'add lecturer', and 'remove lecturer' methods
 
-  _areas = [];
-  _lecturers = []; // Name, surname, position, company, experience, courses, contacts
+  _areas: Area[] = [];
+  _lecturers: lecturerType[] = []; // Name, surname, position, company, experience, courses, contacts
+
+  get areas(): typeof this._areas {
+    return this._areas;
+  }
+
+  get lecturers(): typeof this._lecturers {
+    return this._lecturers;
+  }
 
-  get areas() {
-    this._areas;
+  addArea(area: Area): void {
+    this._areas.push(area);
   }
 
-  get lecturers() {
-    this._lecturers;
+  removeArea(inputArea: Area): void {
+    this._areas = this._areas.filter(area => area._name !== inputArea._name);
+  }
+
+  addLecturer(lecturer: lecturerType): void {
+    this._lecturers.push(lecturer);
+  }
+
+  removeLecturer(inputLecturer: lecturerType): void {
+    this._lecturers = this._lecturers.filter(lecturer => lecturer.name !== inputLecturer.name);
   }
 }
 
 class Area {
   // implement getters for fields and 'add/remove level' methods
-  _levels = [];
-  _name;
+  _levels: Level[] = [];
+  _name: string;
 
-  constructor(name) {
+  constructor(name: string) {
     this._name = name;
   }
+
+  addLevel(level: Level): void {
+    this._levels.push(level);
+  }
+
+  removeLevel(inputLevel: Level): void {
+    this._levels = this._levels.filter(level => level._name !== inputLevel._name);
+  }
 }
 
 class Level {
   // implement getters for fields and 'add/remove group' methods
 
-  _groups;
-  _name;
+  _groups: Group[] = [];
+  _name: string;
+  _description: string;
 
-  constructor(name, description) {
-    this.name = name;
+  constructor(name: string, description: string) {
+    this._name = name;
     this._description = description;
   }
-}
 
+  addGroup(group: Group): void {
+    this._groups.push(group);
+  }
+
+  removeGroup(inputGroup: Group): void {
+    this._groups.filter(group => group.directionName !== inputGroup.directionName);
+  }
+}
 class Group {
   // implement getters for fields and 'add/remove student' and 'set status' methods
 
-  _area;
-  _status;
-  _students = []; // Modify the array so that it has a valid toSorted method*
+  _area: Area;
+  _status: boolean;
+  _students: Student[] = []; // Modify the array so that it has a valid toSorted method*
+  directionName: string;
+  levelName: string;
+
+  set status(status: boolean) {
+    this.status = status;
+  }
 
-  constructor(directionName, levelName) {
+  constructor(directionName: string, levelName: string, area: Area, status: boolean) {
     this.directionName = directionName;
     this.levelName = levelName;
+    this._area = area;
+    this._status = status;
   }
 
-  showPerformance() {
-    const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+  showPerformance(): object[] {
+    const sortedStudents: object[] = this._students.toSorted(
```

Comment:
If you annotate parameters as a Student entity, I think it makes sense to add the same but with a list of Students.